From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ruViolence <78062896+ruViolence@users.noreply.github.com>
Date: Fri, 3 Jun 2022 23:37:07 +0500
Subject: [PATCH] fix: clientside entity desynchronization

Ported from 1.8 to prevent recreating entities that are still within view distance of the player.

diff --git a/src/main/java/net/minecraft/server/EntityTrackerEntry.java b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
index d7fbfad3e64cf043dd5c17474c095988a2833a17..2218b542908688972e23c7c0296617e2975a8eef 100644
--- a/src/main/java/net/minecraft/server/EntityTrackerEntry.java
+++ b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
@@ -505,8 +505,10 @@ public class EntityTrackerEntry {
     }
     private boolean isInRangeOfPlayer(EntityPlayer entityplayer) {
         // Paper end
-        double d0 = entityplayer.locX - (double) this.xLoc / 4096.0D;
-        double d1 = entityplayer.locZ - (double) this.zLoc / 4096.0D;
+        // Reaper start - Fix clientside entity desynchronization
+        double d0 = entityplayer.locX - this.tracker.locX;
+        double d1 = entityplayer.locZ - this.tracker.locZ;
+        // Reaper end
         int i = Math.min(this.e, this.f);
 
         return d0 >= (double) (-i) && d0 <= (double) i && d1 >= (double) (-i) && d1 <= (double) i && this.tracker.a(entityplayer);
